src/psd_tools/api/psd_image.py

```diff
@@ -41,6 +47,48 @@ def __init__(self, data):
         self._tagged_blocks = None
         self._init()
 
+    def buildFromConfigure(self,
+                        images_tree : Dict[str, Union[Dict, np.ndarray]],
+                        configs : Dict[str, Union[Dict, str, int]]) -> None :
+        #TODO. @OFF.Donald
+        '''
+        format of configs :
+            {
+                'commons' : {
+                    #CONTENTS
+                }
+                'layers' : {
+                    'sub_value1':{
+                        'configs' : {
+                            #CONTENTS
+                        }
+                    },
+                    'sub_value2':{
+                        'configs' : {
+                            #CONTENTS
+                        }
+                    }
+                }
+            }
+        '''
+        commons_configs, layers_configs = configs['commons'], configs['layers']
+        layer_stack = [(self, {sub:images_tree[sub]}, {sub:layers_configs[sub]}) for sub in images_tree]
+        parent = self
+
+        while(layer_stack):
+            parent, subtree, sub_configs = layer_stack.pop()
+            for layer_name, layer_item in subtree.items():
+                is_group = True if type(layer_item) is dict else False
+                config = updateConfig(commons=commons_configs,
+                                        configs=  sub_configs[layer_name].pop('configs'))
+                new_layer = buildLayer(configs  = config,
+                                        images  = None if is_group else layer_item,
+                                        psd     = self,
+                                        parent  = parent)
+                parent._layers.append(new_layer)
```

Comment:
You did a great job, Donald.

I have a question. In the buildFromConfigure method, this line only accumulates new_layer. Then where is the code to remove the old existing layer? Or is this your intention to accumulate new layers to psd file?

Reply:
This commit only deals with generating a new PSD file, not modifying (ex. removing the layer in the existing file). Because in our scenario, a new PSD file should be made from the image and configured file. removing existing layer is easy, just pop at PSDImage._layers. if we need the removing layer, then I will add

Reply:
Ok, I got it. Then how about adding code to check whether the 'self' is empty or not?

Reply:
Um... does 'self' become empty? because self is a class pointer, it always has a declared class object (buildFromConfigure is instance class function, not classmethod)

ex. use case
```python
# declare new psd object
new_psd = PSDImage(*args)  
# build layers
new_psd.buildFromConfigure(**kargs)
# save psd file
new_psd.save(PATH)
```
Plz, let me know if I missed anything!

Reply:
Sorry for misleading you. What I mean is that you should check the emptiness of layers of self. I understood your intention, but without context, it is hard to perceive this code is for generating a new psd file. To clarify the purpose of this code, I suggested some layer emptiness checking.

Reply:
Oh, I understand what u meant. 
Following is the example that complements ur suggestion. 

case of force reset
```python
if len(self._layers) != 0 : logging.info['Reset layers of PSD']
self._layers = [] # reset
```

case of raise error
```python
if len(self._layers) != 0:
    raise( Exception:'PSDImage already has be builted, please reset layers frist')
```

Reply:
I think raise error is proper in this case! An unintentional reset might be dangerous.